pilot/pkg/model/service.go

```diff
@@ -310,8 +311,28 @@ type ServiceDiscovery interface {
 	//
 	// Similar concepts apply for calling this function with a specific
 	// port, hostname and labels.
+	// Deprecated: made obsolete by InstancesByPort
 	Instances(hostname Hostname, ports []string, labels LabelsCollection) ([]*ServiceInstance, error)
 
+	// InstancesByPort retrieves instances for a service on the given ports with labels that match
+	// any of the supplied labels. All instances match an empty tag list.
+	//
+	// For example, consider the example of catalog.mystore.com as described in NetworkEndpoints
+	// Instances(catalog.myservice.com, 80) ->
+	//      --> NetworkEndpoint(172.16.0.1:8888), Service(catalog.myservice.com), Labels(foo=bar)
+	//      --> NetworkEndpoint(172.16.0.2:8888), Service(catalog.myservice.com), Labels(foo=bar)
+	//      --> NetworkEndpoint(172.16.0.3:8888), Service(catalog.myservice.com), Labels(kitty=cat)
+	//      --> NetworkEndpoint(172.16.0.4:8888), Service(catalog.myservice.com), Labels(kitty=cat)
+	//
+	// Calling Instances with specific labels returns a trimmed list.
+	// e.g., Instances(catalog.myservice.com, 80, foo=bar) ->
+	//      --> NetworkEndpoint(172.16.0.1:8888), Service(catalog.myservice.com), Labels(foo=bar)
+	//      --> NetworkEndpoint(172.16.0.2:8888), Service(catalog.myservice.com), Labels(foo=bar)
+	//
+	// Similar concepts apply for calling this function with a specific
+	// port, hostname and labels.
+	InstancesByPort(hostname Hostname, ports []int, labels LabelsCollection) ([]*ServiceInstance, error)
```

Comment:
Is there any use case for multiple ports ? Never found one...

Reply:
Me neither. Carried over from old code. might as well make it single